content/articles/implementation-of-fuzzy-logic-in-matlab/index.md

```diff
@@ -0,0 +1,121 @@
+```matlab
+%This is not a real code but just to make you understand
+if credit >= 750
+return credit is good
+else
+credit is medium
+```
+The above is a logical statement. It means that it is either entirely true or not. The equation that we have set up makes this problem easy to solve, but the issue is the linguistic term that we use to describe the degree of something is fuzzy; that is, language is not precise. Making this language precise, we lose some intent. How?, Now a score of 750 might be considered good, very good or neutral to different bankers. So if we want to encode the knowledge to the banking industry, we need to consider the fuzziness, which is where the fuzzy logic comes in.
+
+![fuzzy concept](/engineering-education/implementation-of-fuzzy-logic-in-matlab/fuzzy_one.png)
+*Figure1.0 Flow chart for the fuzzy logic*
+
+As shown in the diagram above, you convert the input with precise value with a particular rating to fuzzy variables with fuzzy logic. This process is known as fuzzification. Then, you apply the fuzzy logic rules to the fuzzy variables to create a new fuzzy number. Finally, you convert the fuzzy number back to a crisp, specific output in a process called fuzzification.
+
+### Fuzzy logic approach
+Suppose we poll 100 bankers and ask them to assign the transition from bad to neutral credit and change from neutral to good. More than likely, in this case, there are going to be a different value for this. Let us assume that every banker thought 750 and above is good credit, but only half thought 700 was good, and none thought 650 was good. It traces out the particular membership function for good credit. We can do the same for neutral credit and bad credit. Here, we will end up with three overlapping membership functions.
+
+![fuzzy logic](/engineering-education/implementation-of-fuzzy-logic-in-matlab/fuzzy_two.png)
+*Figure2.0 The basic fuzzy concept*
+
+Now, we can see that the statement's truth does not belong to a single set but can belong to multiple sets based on the degree of the truth.
+
+### Implementation of the fuzzy logic in Matlab
+Consider a very simple example;
+We need to control the speed of a motor by changing the input voltage when a set point is defined. If the motor runs fast for some reason, we need to slow it down by reducing the input voltage. Conversely, if the motor slows below the set point, the input voltage must be increased so that the motor reaches the set point.
+Below are the description for the inputs. These will help in defining the rules.
+`Too slow` 
+Just right
+Too fast
+The statements are self-explanatory.
```

Comment:
Yes totally.

The way of putting them together is not right.

As a beginner, the reader would find it difficult to comprehend on what you're trying to accomplish.

Have a look at few of my articles [here](https://www.section.io/engineering-education/authors/srishilesh-p-s/) on how I've explained code tutorials